plugins/Redirector/class.redirector.plugin.php

```diff
@@ -602,18 +602,24 @@ public static function topicFilter(&$get) {
                 '_arg2' => 'Page'
             ];
         } else {
-            // This is an ipb style topic.
-            return [
-                'p' => 'CommentID',
-                '_arg0' => [
-                    'DiscussionID',
-                    'Filter' => [__CLASS__, 'removeID'],
-                ],
-                '_arg1' => [
-                    'Page',
-                    'Filter' => [__CLASS__, 'IPBPageNumber'],
-                ],
-            ];
+            if (!isset($get['_arg3'])) {
```

Comment:
I understand that this might work for PunBB.

My main concern here was mostly for IPB migrations. You base your check simply on the fact that the 3rd parameter is not set to determine if it's either a PunBB or a IPB URL.

The example you gave in the last PR has a page number which means the 3rd parameter will be set and in this case the page number will be retrieved from the Url.

However, what happens if there is no page number in the Url as in the example below?

topic/403777-40-permalinkurl-structure-options/

This will be considered as a PunBB url when in fact, it's an IPB url and `403777-40-permalinkurl-structure-options` will be returned as the discussion id.

Here is the result you will get as an example.

![Screen Shot 2020-03-16 at 3 14 20 PM](https://user-images.githubusercontent.com/5429361/76792435-6fb4d280-6799-11ea-88ab-d4cf26b8b953.png)

Reply:
Oh I see, sorry I never actually see how IPB Url looks like for discussion, and lack of consider for IPB. Thank to point it out.

Reply:
@patches , and also another thing i will like to confirm. Do you think the example I found will be the standard PunBB discussion URL with page number? 

punbb example with page number:
```
http://punbb.informer.com/forums/topic/10778/what-antivirus-do-you-use/page/2/
```
If it is , the page number should be at arg position 5? like **'_arg5'**?